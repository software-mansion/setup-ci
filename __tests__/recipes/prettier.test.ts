import { readFileSync, existsSync } from 'fs'
import { join } from 'path'
import {
  cli,
  getPackageJsonWithoutVersions,
  PULL_REQUEST_FLAG,
  removeTestProject,
  setupTestProject,
  TEST_PROJECTS,
} from '../utils'

const FLAG = 'prettier'

describe('prettier check recipe', () => {
  afterEach(removeTestProject)

  const PROJECTS = [
    'rn-setup-ci-yarn-flat',
    'rn-setup-ci-npm-flat',
    'rn-setup-ci-yarn-monorepo',
    'rn-setup-ci-npm-monorepo',
    'rn-setup-ci-create-expo-stack',
    'rn-setup-ci-create-expo-stack-bun',
    'rn-setup-ci-create-expo-stack-pnpm',
  ]

  for (const projectName of PROJECTS) {
    test(`${projectName} prettier check`, async () => {
      setupTestProject(projectName)
      const {
        packageManager,
        repoRoot,
        appRoot,
        workflowNamePrefix,
        existingConfig,
      } = TEST_PROJECTS[projectName]

      const output = await cli([PULL_REQUEST_FLAG, FLAG], {
        cwd: appRoot,
      })

      for (const message of [
        `Detected ${packageManager} as your package manager.`,
        'Configuring project for Prettier check',
        'Created Prettier check workflow for events: [pull_request]',
      ]) {
        expect(output).toContain(message)
      }

      const packageJson = await getPackageJsonWithoutVersions(
        join(appRoot, 'package.json')
      )
      expect(packageJson).toMatchSnapshot()

      const filesToCheck = [
        join(
          repoRoot,
          '.github',
          'workflows',
          `${workflowNamePrefix}prettier.yml`
        ),
      ]

      if (!existingConfig?.prettier) {
        filesToCheck.push(join(appRoot, '.prettierrc'))
        filesToCheck.push(join(appRoot, '.prettierignore'))
      } else if (existingConfig.prettier !== '.prettierrc') {
        expect(existsSync(join(appRoot, '.prettierrc'))).toBe(false)
      }

      filesToCheck.forEach((file) => {
        const fileContent = readFileSync(file).toString()
        expect(fileContent).toMatchSnapshot()
      })
    })
  }
})
