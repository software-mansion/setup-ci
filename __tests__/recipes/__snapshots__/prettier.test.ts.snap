// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier check recipe rn-setup-ci-create-expo-stack prettier check 1`] = `
Object {
  "dependencies": Object {
    "react": "",
    "react-native": "",
  },
  "devDependencies": Object {
    "@babel/core": "",
    "@babel/preset-env": "",
    "@babel/runtime": "",
    "@react-native/babel-preset": "",
    "@react-native/metro-config": "",
    "@react-native/typescript-config": "",
    "@types/react": "",
    "@types/react-test-renderer": "",
    "prettier": "",
    "react-test-renderer": "",
  },
  "engines": Object {
    "node": ">=18",
  },
  "name": "RnSetupCiYarnFlat",
  "packageManager": "yarn@3.6.4",
  "private": true,
  "scripts": Object {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "prettier:check": "prettier --check \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "prettier:write": "prettier --write \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "start": "react-native start",
  },
  "version": "0.0.1",
}
`;

exports[`prettier check recipe rn-setup-ci-create-expo-stack prettier check 2`] = `
"name: Prettier check

on:
  pull_request:

jobs:
  prettier-check:
    name: Prettier check
    runs-on: ubuntu-latest
    steps:
      - name: 🏗 Setup repo
        uses: actions/checkout@v4

      - name: 🌿 Setup Node
        uses: actions/setup-node@v4
        with:
          node-version-file: '.nvmrc'
          cache: 'npm'

      - name: 📦 Install dependencies
        run: npm ci

      - name: ✨ Run Prettier check
        run: npm run prettier:check
"
`;

exports[`prettier check recipe rn-setup-ci-create-expo-stack-bun prettier check 1`] = `
Object {
  "dependencies": Object {
    "react": "",
    "react-native": "",
  },
  "devDependencies": Object {
    "@babel/core": "",
    "@babel/preset-env": "",
    "@babel/runtime": "",
    "@react-native/babel-preset": "",
    "@react-native/metro-config": "",
    "@react-native/typescript-config": "",
    "@types/react": "",
    "@types/react-test-renderer": "",
    "prettier": "",
    "react-test-renderer": "",
  },
  "engines": Object {
    "node": ">=18",
  },
  "name": "RnSetupCiYarnFlat",
  "packageManager": "yarn@3.6.4",
  "private": true,
  "scripts": Object {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "prettier:check": "prettier --check \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "prettier:write": "prettier --write \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "start": "react-native start",
  },
  "version": "0.0.1",
}
`;

exports[`prettier check recipe rn-setup-ci-create-expo-stack-bun prettier check 2`] = `
"name: Prettier check

on:
  pull_request:

jobs:
  prettier-check:
    name: Prettier check
    runs-on: ubuntu-latest
    steps:
      - name: 🏗 Setup repo
        uses: actions/checkout@v4

      - name: 🥟 Setup Bun
        uses: oven-sh/setup-bun@v2
        with:
          bun-version-file: '.bun-version'

      - name: 📦 Install dependencies
        run: bun install --frozen-lockfile

      - name: ✨ Run Prettier check
        run: bun run prettier:check
"
`;

exports[`prettier check recipe rn-setup-ci-npm-flat prettier check 1`] = `
Object {
  "dependencies": Object {
    "react": "",
    "react-native": "",
  },
  "devDependencies": Object {
    "@babel/core": "",
    "@babel/preset-env": "",
    "@babel/runtime": "",
    "@react-native/babel-preset": "",
    "@react-native/metro-config": "",
    "@react-native/typescript-config": "",
    "@types/react": "",
    "@types/react-test-renderer": "",
    "prettier": "",
    "react-test-renderer": "",
  },
  "engines": Object {
    "node": ">=18",
  },
  "name": "RnSetupCiYarnFlat",
  "packageManager": "yarn@3.6.4",
  "private": true,
  "scripts": Object {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "prettier:check": "prettier --check \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "prettier:write": "prettier --write \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "start": "react-native start",
  },
  "version": "0.0.1",
}
`;

exports[`prettier check recipe rn-setup-ci-npm-flat prettier check 2`] = `
"name: Prettier check

on:
  pull_request:

jobs:
  prettier-check:
    name: Prettier check
    runs-on: ubuntu-latest
    steps:
      - name: 🏗 Setup repo
        uses: actions/checkout@v4

      - name: 🌿 Setup Node
        uses: actions/setup-node@v4
        with:
          node-version-file: 'package.json'
          cache: 'npm'

      - name: 📦 Install dependencies
        run: npm ci

      - name: ✨ Run Prettier check
        run: npm run prettier:check
"
`;

exports[`prettier check recipe rn-setup-ci-npm-flat prettier check 3`] = `
"{
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"trailingComma\\": \\"es5\\"
}
"
`;

exports[`prettier check recipe rn-setup-ci-npm-flat prettier check 4`] = `
"node_modules
dist
build
"
`;

exports[`prettier check recipe rn-setup-ci-npm-monorepo prettier check 1`] = `
Object {
  "dependencies": Object {
    "@expo/vector-icons": "",
    "@react-navigation/native": "",
    "expo": "",
    "expo-constants": "",
    "expo-font": "",
    "expo-linking": "",
    "expo-router": "",
    "expo-splash-screen": "",
    "expo-status-bar": "",
    "expo-system-ui": "",
    "expo-updates": "",
    "expo-web-browser": "",
    "react": "",
    "react-dom": "",
    "react-native": "",
    "react-native-gesture-handler": "",
    "react-native-reanimated": "",
    "react-native-safe-area-context": "",
    "react-native-screens": "",
    "react-native-web": "",
  },
  "devDependencies": Object {
    "@babel/core": "",
    "@types/react": "",
    "@types/react-test-renderer": "",
    "jest-expo": "",
    "prettier": "",
    "react-test-renderer": "",
  },
  "main": "expo-router/entry",
  "name": "expo-app",
  "private": true,
  "scripts": Object {
    "android": "expo start --android",
    "ios": "expo start --ios",
    "prettier:check": "prettier --check \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "prettier:write": "prettier --write \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "reset-project": "node ./scripts/reset-project.js",
    "start": "expo start",
    "web": "expo start --web",
  },
  "version": "1.0.0",
}
`;

exports[`prettier check recipe rn-setup-ci-npm-monorepo prettier check 2`] = `
"name: Prettier check

on:
  pull_request:
    paths:
      - apps/expo-app/**

jobs:
  prettier-check:
    name: Prettier check
    runs-on: ubuntu-latest
    steps:
      - name: 🏗 Setup repo
        uses: actions/checkout@v4

      - name: 🌿 Setup Node
        uses: actions/setup-node@v4
        with:
          node-version-file: 'apps/expo-app/.nvmrc'
          cache: 'npm'

      - name: 📦 Install dependencies
        run: npm ci

      - name: ✨ Run Prettier check
        run: npm run prettier:check
        working-directory: apps/expo-app
"
`;

exports[`prettier check recipe rn-setup-ci-npm-monorepo prettier check 3`] = `
"{
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"trailingComma\\": \\"es5\\"
}
"
`;

exports[`prettier check recipe rn-setup-ci-npm-monorepo prettier check 4`] = `
"node_modules
dist
build
"
`;

exports[`prettier check recipe rn-setup-ci-yarn-flat prettier check 1`] = `
Object {
  "dependencies": Object {
    "react": "",
    "react-native": "",
  },
  "devDependencies": Object {
    "@babel/core": "",
    "@babel/preset-env": "",
    "@babel/runtime": "",
    "@react-native/babel-preset": "",
    "@react-native/metro-config": "",
    "@react-native/typescript-config": "",
    "@types/react": "",
    "@types/react-test-renderer": "",
    "prettier": "",
    "react-test-renderer": "",
  },
  "engines": Object {
    "node": ">=18",
  },
  "name": "RnSetupCiYarnFlat",
  "packageManager": "yarn@3.6.4",
  "private": true,
  "scripts": Object {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "prettier:check": "prettier --check \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "prettier:write": "prettier --write \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "start": "react-native start",
  },
  "version": "0.0.1",
}
`;

exports[`prettier check recipe rn-setup-ci-yarn-flat prettier check 2`] = `
"name: Prettier check

on:
  pull_request:

jobs:
  prettier-check:
    name: Prettier check
    runs-on: ubuntu-latest
    steps:
      - name: 🏗 Setup repo
        uses: actions/checkout@v4

      - name: 🌿 Setup Node
        uses: actions/setup-node@v4
        with:
          node-version-file: 'package.json'
          cache: 'yarn'

      - name: 📦 Install dependencies
        run: yarn install --immutable

      - name: ✨ Run Prettier check
        run: yarn run prettier:check
"
`;

exports[`prettier check recipe rn-setup-ci-yarn-flat prettier check 3`] = `
"{
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"trailingComma\\": \\"es5\\"
}
"
`;

exports[`prettier check recipe rn-setup-ci-yarn-flat prettier check 4`] = `
"node_modules
dist
build
"
`;

exports[`prettier check recipe rn-setup-ci-yarn-monorepo prettier check 1`] = `
Object {
  "dependencies": Object {
    "@expo/vector-icons": "",
    "@react-navigation/native": "",
    "expo": "",
    "expo-constants": "",
    "expo-font": "",
    "expo-linking": "",
    "expo-router": "",
    "expo-splash-screen": "",
    "expo-status-bar": "",
    "expo-system-ui": "",
    "expo-updates": "",
    "expo-web-browser": "",
    "react": "",
    "react-dom": "",
    "react-native": "",
    "react-native-gesture-handler": "",
    "react-native-reanimated": "",
    "react-native-safe-area-context": "",
    "react-native-screens": "",
    "react-native-web": "",
  },
  "devDependencies": Object {
    "@babel/core": "",
    "@types/react": "",
    "@types/react-test-renderer": "",
    "jest-expo": "",
    "prettier": "",
    "react-test-renderer": "",
  },
  "main": "expo-router/entry",
  "name": "expo-app",
  "private": true,
  "scripts": Object {
    "android": "expo start --android",
    "ios": "expo start --ios",
    "prettier:check": "prettier --check \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "prettier:write": "prettier --write \\"**/*.{ts,tsx,js,jsx,json,css,scss,md}\\"",
    "reset-project": "node ./scripts/reset-project.js",
    "start": "expo start",
    "web": "expo start --web",
  },
  "version": "1.0.0",
}
`;

exports[`prettier check recipe rn-setup-ci-yarn-monorepo prettier check 2`] = `
"name: Prettier check

on:
  pull_request:
    paths:
      - apps/expo-app/**

jobs:
  prettier-check:
    name: Prettier check
    runs-on: ubuntu-latest
    steps:
      - name: 🏗 Setup repo
        uses: actions/checkout@v4

      - name: 🌿 Setup Node
        uses: actions/setup-node@v4
        with:
          node-version-file: 'apps/expo-app/.nvmrc'
          cache: 'yarn'

      - name: 📦 Install dependencies
        run: yarn install --immutable

      - name: ✨ Run Prettier check
        run: yarn run prettier:check
        working-directory: apps/expo-app
"
`;

exports[`prettier check recipe rn-setup-ci-yarn-monorepo prettier check 3`] = `
"{
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"trailingComma\\": \\"es5\\"
}
"
`;

exports[`prettier check recipe rn-setup-ci-yarn-monorepo prettier check 4`] = `
"node_modules
dist
build
"
`;
